### ISSUE TITLE
`mc mirror --watch` is not working when object deleted from source

### ISSUE THREAD

**Issue by AhmedAbouelkher** [minio/mc#4379]
## Expected behavior
Deleting an object from minio is reflected on the destination AWS s3 in my case.

## Actual behavior
Creation mirroring works, but not deletion with the same exact command session.

## Steps to reproduce the behavior
- Remove an object while the command is running.

## mc --version
RELEASE.2022-11-07T23-47-39Z

## System information
OS: `Debian GNU/Linux 11 (bullseye)`

## Related topic 
I posted a detailed post in the Q&A sections regarding the same topic two days ago. [Here](https://github.com/minio/minio/discussions/16132)

**Comment by vadmeste** [comment]
@AhmedAbouelkher this works fine for you, can you try to describe how to reproduce with more detailed steps ?

**Comment by AhmedAbouelkher** [comment]
> @AhmedAbouelkher this works fine for you, can you try to describe how to reproduce with more detailed steps ?

I didn't do anything unusual, just added the aliases and ran the command. There are no more steps regarding that.

I've confirmed that the alias data is valid both for S3 and minio.

**Comment by vadmeste** [comment]
@AhmedAbouelkher then it is possible that the deletion is not occuring in the destination bucket, can you do `mc mirror` with --debug flag and upload the output here ?

**Comment by AhmedAbouelkher** [comment]
Do you mean the source, right? 

Uploaded `screen.png` to the source using this command

```bash
mc mirror --watch myminio/tst s3/rraw-media-upload --remove --debug
```

![Screen Shot 2022-11-27 at 11 31 36 PM](https://user-images.githubusercontent.com/56558577/204160729-c6e5baca-28dc-45dd-a446-e15b65935c88.png)

When removing `screen.png` there are no logs at all from the `mc` command

**Comment by vadmeste** [comment]
@AhmedAbouelkher strange, can you try to upload an object and checks if it is replicated without mc mirror command is already running ?

**Comment by AhmedAbouelkher** [comment]
I created a new bucket on minio (named `tst2`) and tried to sync with AWS S3 and deletion **worked** without any issue
command
```bash
mc mirror --watch myminio/tst2 s3/media-ddata --debug
```
the previous bucket that I was testing with (named `tst`) still has not worked deletion for some reason.
I think that issue might be that `tst` has a bucket replication process sat up with another minio site, but I'm not sure.

I will close this for now, due to the conflict results I got.
